sqlglot/dialects/snowflake.py

```diff
@@ -324,6 +324,10 @@ def intersect_op(self, expression):
                 self.unsupported("INTERSECT with All is not supported in Snowflake")
             return super().intersect_op(expression)
 
+        def min_sql(self, expression):
+            name = "LEAST" if expression.expressions else "MIN"
+            return rename_func(name)(self, expression)
```

Comment:
I think other dialects can leverage this too, e.g. postgres seems to use the same semantics as snowflake.